Add support for psp

otnielvh opened the pull request:
This PR adds the yaml file needed for running nginx when port security policy (PSP) is enabled on the Kubernetes cluster. Me and my colleagues struggled with it somewhat, and felt it would be nice to have it together with the rest of nginx.

aledbf commented:
@otnielvh thank you for your contribution. Please move the doc and yaml files to the example section.
The reason for this is related to the (subjective) definition of what should be defined in a PSP.

aledbf commented:
Also, it would be great if you can add an e2e test using PSP to avoid regressions in the future.

otnielvh commented:
@aledbf, thank you for your feedback. I moved the files into the example section, however I'm not sure how to address e2e testing.  AFAIK enabling PSP requires changes in the cluster that are not supported via kubectl (I'm running in GKE and use gcloud). From a brief look the other tests don't require such fundamental changes. Could you please point me in the direction of how to write a test that changes the cluster not through kubectl or client api? If you can recall any examples that would be a great help. Thank you again for you time.

aledbf commented:
@otnielvh I forgot I already wrote one :man_facepalming: https://github.com/kubernetes/ingress-nginx/blob/master/test/e2e/settings/pod_security_policy.go
Please squash the commits

aledbf commented:
/ok-to-test

aledbf commented:
Also, please add a link to this new example in this section https://github.com/kubernetes/ingress-nginx/blob/master/mkdocs.yml#L66

diazjf commented on docs/examples/psp/README.md:
```diff
@@ -0,0 +1,22 @@
+
+Note: PSP permissions must be granted before to the creation of the Deployment and the ReplicaSet.
+If the Deployment or ReplicaSet already exist, they will receive the PSP permissions
+only after deleting them and reapplying mandatory.yaml.
```
Can you link mandatory.yaml

diazjf commented on docs/examples/psp/README.md:
```diff
@@ -0,0 +1,22 @@
+Kubernetes however provides a more fine-grained authorization policy called 
+Pod Security Policy (PSP).
+
+PSP allows the cluster owner to define the permission of each object, for example creating a pod.
```
Add link to https://kubernetes.io/docs/concepts/policy/pod-security-policy/

otnielvh commented:
Squashed the commits and added the links mentioned. I still have two more points:
1. @aledbf  the link  https://github.com/kubernetes/ingress-nginx/blob/master/mkdocs.yml#L66 points to `third party addons` section, I assume this is a typo and PSP should be in examples, I'm just want to double check.
2. @diazjf, I started off by adding some documentation in the main docs, but then removed it because the first thing a user sees is some PSP stuff that is quite advanced and most people don't use. Where do you think would be a good place to link to the PSP documentation?

otnielvh commented:
/restart

otnielvh commented:
/retest

otnielvh commented:
/test pull-ingress-nginx-test

aledbf commented:
/lgtm

aledbf commented:
@otnielvh thanks!
